.buildkite/scripts/steps/cloud/build_and_deploy.sh

```diff
@@ -59,11 +61,10 @@ if [ -z "${CLOUD_DEPLOYMENT_ID}" ]; then
 
   retry 5 5 vault write "secret/kibana-issues/dev/cloud-deploy/$CLOUD_DEPLOYMENT_NAME" username="$CLOUD_DEPLOYMENT_USERNAME" password="$CLOUD_DEPLOYMENT_PASSWORD"
 else
-  ecctl deployment show "$CLOUD_DEPLOYMENT_ID" --generate-update-payload | jq '
-    .resources.kibana[0].plan.kibana.docker_image = "'$CLOUD_IMAGE'" |
-    .resources.kibana[0].plan.kibana.version = "'$VERSION'" |
-    .resources.elasticsearch[0].plan.elasticsearch.version = "'$VERSION'"
-    ' > /tmp/deploy.json
+ecctl deployment show "$CLOUD_DEPLOYMENT_ID" --generate-update-payload | jq '
+  .resources.kibana[0].plan.kibana.docker_image = "'$CLOUD_IMAGE'" |
+  (.. | select(.version? != null).version) = "'$VERSION'"
```

Comment:
This is intended for PR's across version upgrades.  It recursively updates all version references to the same version used in the Kibana docker image.